### PR TITLE
lib/cliutil: create package for exporting urfave/cli utilities

### PR DESCRIPTION
**Pull request by bobheadxi**
Move `sg`'s docgen code into `lib/cliutil/docgen` for sharing with the `mi` CLI. In the future we can move other useful utilities for `urfave/cli` and CLIs in general into this set of subpackages for use in `mi`.

For https://github.com/sourcegraph/controller/pull/139

## Test plan

CI

### CODE REVIEW

**Review by michaellzc — APPROVED** [review]
❤️

**Comment by michaellzc** [comment]
should `lib/cliutil` be its own go module? otherwise if people just want `lib/errors`, they will be pulling in the urfave modules as well, it feels kind of heavy

**Comment by bobheadxi** [comment]
> should lib/cliutil be its own go module? otherwise if people just want lib/errors, they will be pulling in the urfave /cli related modules as well, it feels kind of heavy

Hmm I think for ease of operation we have just been allowing `lib` grow somewhat, it already has a pretty extensive dependency tree. I think if we want to do this later we can just add a `go.mod` in there and update dependents

**Comment by bobheadxi** [comment]
`urfave/cli` is a pretty lightweight dependency: https://github.com/urfave/cli/blob/main/go.mod

File: dev/sg/sg_help.go [hunk 1]
```diff
@@ -7,19 +7,20 @@ import (
...
 var helpCommand = &cli.Command{
-	Name:      "help",
-	ArgsUsage: " ", // no args accepted for now
-	Usage:     "Get help and docs about sg",
-	Category:  CategoryUtil,
+	Name:            "help",
+	ArgsUsage:       " ", // no args accepted for now
+	Usage:           "Get help and docs about sg",
+	Category:        CategoryUtil,
+	HideHelpCommand: true, // we don't want a "sg help help" :)
```
Comment (bobheadxi) [inline dev/sg/sg_help.go:23]: Caused by library upgrade it seems

File: doc/dev/background-information/sg/reference.md [hunk 2]
```diff
@@ -723,7 +723,7 @@ Available schemas:
 
 Flags:
 
-* `--db="<value>"`: The target `schema(s)` to validate. Comma-separated values are accepted. Supply "all" to validate all schemas. (default: [all])
+* `--db="<value>"`: The target `schema(s)` to validate. Comma-separated values are accepted. Supply "all" to validate all schemas. (default: "all")
```
Comment (bobheadxi) [inline doc/dev/background-information/sg/reference.md:726]: also caused by library upgrade

**Review by kalanchan — APPROVED** [review]

File: dev/sg/sg_lint.go [hunk 3]
```diff
@@ -70,7 +70,7 @@ sg lint --help
 		// If more than 1 target is requested, hijack subcommands by setting it to nil
 		// so that the main lint command can handle it the run.
 		if cmd.Args().Len() > 1 {
-			cmd.App.Commands = nil
+			cmd.Command.Subcommands = nil
```
Comment (bobheadxi) [inline dev/sg/sg_lint.go:73]: _also_ seems caused by library upgrade. I think the old behaviour of having to reach into `cmd.App` was a bug

**Pull request merged**